uflash.py

```diff
@@ -54,18 +61,17 @@ def hexlify(script):
     script = script.replace(b'\r', b'\n')
     # Add header, pad to multiple of 16 bytes.
     data = b'MP' + struct.pack('<H', len(script)) + script
-    data = data + bytes(16 - len(data) % 16)
```

Comment:
This bit was fun:
`bytes()` on Python2.7 is a compatibility shim and just an alias to `str()`
For normal uses of `bytes()` that would be fine, as `str()` in Python 2 is ASCII and therefore bytes anyway.
However this line was using a special convenience feature of the Python 3 real `bytes` constructor, where if you pass it an int instead of a string, which returns a string of `null` bytes of that length.

So instead of padding the last chunk of the data out to 16 bytes, it just appended the number of bytes it should have been padding as a string.

Py3>>> bytes(4)
b`\x00\x00\x00\x00'
Py2>>> bytes(4)
'4'